packages/ui/lib/components/list-item-action/ListItemAction.tsx

```diff
@@ -13,9 +13,14 @@ import {
 
 import { Icon, IconType } from "../icon";
 
+type BaseBadgeProps = Pick<ChipProps, "color" | "label" | "size">;
+
 export type BadgeProps = {
-  icon: IconType;
-} & Pick<ChipProps, "color" | "label" | "size">;
+  icon?: IconType;
+} & BaseBadgeProps;
+
+export type FullBadgeProps = Omit<BadgeProps, "icon"> &
+  Required<Pick<BadgeProps, "icon">>;
```

Comment:
is it useful?

Reply:
For now, we found this useful to avoid certain type errors across apps badge configurations, since the`icon` prop in `badgeProps` has become optional.